public/app/plugins/panel/table/PageSizeEditor.tsx

```diff
@@ -0,0 +1,22 @@
+import { StandardEditorProps } from '@grafana/data';
+import { NumberValueEditor } from '@grafana/ui';
+import React from 'react';
+
+export function PageSizeEditor({ onChange, value, id, context }: StandardEditorProps<number>) {
+  const changeValue = (newValue: number | undefined) => {
+    if (newValue) {
+      context.options.footer.show = false;
+    }
+    onChange(newValue);
+  };
```

Comment:
This mutation... Nothing else comes to my mind other than moving the `pageSize` option under the `footer` property to avoid that. Think this ain't a bad idea given the pagination is actually a footer element - wdyt @zoltanbedi ?

Reply:
So I've changed this to be under the footer property but I'm not sure how can this be not a mutation. Let's talk about this.

Reply:
@zoltanbedi  - I wasn't clear I think. It would help IF we had a custom editor for the entire `footer` property - otherwise the editor only gets the value that's stored under a given option path.